hw/hw13/trigrams.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+import random
+
+open_file = open('sherlock.txt', 'r')
+text = open_file.read()
+
+open_file.close()
+
+#  remove unwanted characters
+for char in ["\\", '(', ')', '-', '.', ';']:
+    if char in text:
+        text = text.replace(char, '')
+for char in ['\n', '--', '-']:
+    if char in text:
+        text = text.replace(char, ' ')
+
+text = text.split()
+
+
+trigrams = {}
+
+
+def words(text):
+    #  add words to dictionary in trigrams
+    for(i, word) in enumerate(text):
+        if i in range(len(text) - 2):
+            trigrams[(text[i], text[i + 1])] = [text[i + 2]]
```

Comment:
This doesn't work. What is happening is when a word pair tuple is found in your trigrams dictionary that already exists, it is replacing the corresponding value with your new `[text[i + 2]]`. Python has a dictionary method called `setdefault` that will set a value for a key pair to a default, or return the value if it already exists. You could do something like this:
```
tragrams.setdefault((text[i], text[i+1]), []).append(text[i+2])
```